src/main/resources/db/changelog/changelog.sql

```diff
@@ -267,3 +267,46 @@ CREATE INDEX `Accounts-SynapseUserId-Index` ON `Accounts`(`synapseUserId`);
 ALTER TABLE `AccountRoles`
 MODIFY COLUMN `role` enum('DEVELOPER','RESEARCHER','ADMIN','TEST_USERS','WORKER','SUPERADMIN') NOT NULL;
 
+-- changeset bridge:13
+
+CREATE TABLE `Assessments` (
+  `guid` varchar(255) NOT NULL,
+  `appId` varchar(255) NOT NULL,
+  `identifier` varchar(255) NOT NULL,
+  `revision` int(10) unsigned NOT NULL,
+  `ownerId` varchar(255),
+  `title` varchar(255) NOT NULL,
+  `summary` text,
+  `validationStatus` text,
+  `normingStatus` text,
+  `osName` varchar(255) NOT NULL,
+  `originGuid` varchar(255),
+  `customizationFields` text,
+  `links` text,
+  `createdOn` bigint(20) unsigned DEFAULT NULL,
+  `modifiedOn` bigint(20) unsigned DEFAULT NULL,
+  `deleted` tinyint(1) DEFAULT '0',
+  `version` bigint(10) unsigned NOT NULL DEFAULT 0,
+  PRIMARY KEY (`guid`),
+  UNIQUE KEY (`appId`, `identifier`, `revision`),
+  CONSTRAINT FOREIGN KEY (`originGuid`) REFERENCES `Assessments` (`guid`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
+
+CREATE INDEX `Assessments-AppId` ON `Assessments`(`appId`);
+CREATE INDEX `Assessments-AppId-Guid` ON `Assessments`(`appId`, `guid`);
+CREATE INDEX `Assessments-AppId-Identifier` ON `Assessments`(`appId`, `identifier`);
+-- the appId-identifier-revision combo is indexed by UNIQUE KEY
+
+CREATE TABLE `Tags` (
+  `value` varchar(255) NOT NULL,
+  `category` varchar(255),
+  PRIMARY KEY (`value`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
```

Comment:
I pulled the category out due to the design documents and the way they organize the assessments by category. These seems to just be a known set of categories (to designers), so they model similar to tags. 

As an alternative, we could include them in tags and return the tag object with the category as part of the tag set. So API consumers could sort the tags by the kind or namespace of the tag (finding the categories, for example). This fits with the proposed /v3/tags API which was going to return them by their category.

These fields can be null (so far as I know, assessments start out untagged).

Let's discuss how to implement in person.

Reply:
Considering this, I have 1) removed category as a special set of tags (this can be accomplished by simply namespacing tags), then 2) this made it tractable to move the tag filtering into the query and out of the code, as you mention below. So that has been fixed as well.